logic/src/commonMain/kotlin/com/wire/kalium/logic/feature/conversation/createconversation/CreateChannelUseCase.kt

```diff
@@ -0,0 +1,32 @@
+/*
+ * Wire
+ * Copyright (C) 2025 Wire Swiss GmbH
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see http://www.gnu.org/licenses/.
+ */
+package com.wire.kalium.logic.feature.conversation.createconversation
+
+import com.wire.kalium.logic.data.conversation.ConversationOptions
+import com.wire.kalium.logic.feature.conversation.createconversation.CreateGroupConversationUseCase.Result
+import com.wire.kalium.logic.data.user.UserId
+
+/**
+ * Use case to create a channel.
+ */
+class CreateChannelUseCase internal constructor(
+    private val createGroupConversation: CreateGroupConversationUseCase
+) {
+    suspend operator fun invoke(name: String, userIdList: List<UserId>, options: ConversationOptions): Result =
+        createGroupConversation.invoke(name, userIdList, options.copy(groupType = ConversationOptions.GroupType.CHANNEL))
```

Comment:
question: I wonder why create a new use case and add tests to it when we can just reuse the same old one

Reply:
For channels, in future we could check for permission first

Reply:
permissions will probably be checked in the UI and hide btn if not possible, in this case i suggest for now we reuse the create conversation instead of passing ConversationOptions and then over writing it in the use case

Reply:
Channels also have different (extra) parameters when creating them. Like whether the channel is public/private, and if it is private, if regular users can add people or not, etc.

Just because we _can_ check in the UI, doesn't mean we should _only_ check in the UI 😄

----

I think it would be cleaner to have a dedicated class that both UseCases use, instead of one UseCase calling the other.

Reply:
It adds a clear separation between them in terms of responsibility and maintenance.

If we need to change something _specific_ about the creation of Regular groups, we change it in the UseCase.

If we need to change something about the _generic_ creation of _any group conversation_, we change it in this `GroupConversationCreator` class.

----

Also, we can inspect the code and clearly understand that "Ok, regular groups have no extra logic to them".